src/include/sph/findNeighbors.hpp

```diff
@@ -0,0 +1,65 @@
+#pragma once
+
+#include <vector>
+#include "Task.hpp"
+
+namespace sphexa
+{
+namespace sph
+{
+
+template <typename T, class Dataset>
+void findNeighborsImpl(Task &t, Dataset &d, const Octree<T> &o)
```

Comment:
Just for the sake of consistency, output parameters should be the last arguments